src/js/services/updateUrlParams.js

```diff
@@ -0,0 +1,16 @@
+const updateUrlParams = (newParams, callback) => {
+  const url = new URL(window.location);
+  const params = new URLSearchParams(url.search);
+
+  Object.keys(newParams).forEach((key) => {
+    params.set(key, newParams[key]);
+  });
+
+  url.search = params.toString();
+  window.history.pushState({}, '', url);
+
+  const paramsObj = Object.fromEntries(params.entries());
+  callback(paramsObj);
```

Comment:
We don't need to call callback here, it's called in apiCallback call in handleLocation, which will be called on click in route function, updating params and pushing them to state goes there, please see comment above